paddle/benchmark_convnet_model.py

```diff
@@ -0,0 +1,197 @@
+    pool1 = fluid.layers.pool2d(
+        input=conv1, pool_type='avg', pool_size=3, pool_stride=2)
+    res1 = layer_warp(block_func, pool1, 64, stages[0], 1)
+    res2 = layer_warp(block_func, res1, 128, stages[1], 2)
+    res3 = layer_warp(block_func, res2, 256, stages[2], 2)
+    res4 = layer_warp(block_func, res3, 512, stages[3], 2)
+    pool2 = fluid.layers.pool2d(
+        input=res4,
+        pool_size=7,
+        pool_type='avg',
+        pool_stride=1,
+        global_pooling=True)
+    out = fluid.layers.fc(input=pool2, size=class_dim, act='softmax')
+    return out
+
+
+def run_benchmark(model, args):
+    if args.use_cprof:
+        pr = cProfile.Profile()
+        pr.enable()
+    start_time = time.time()
+
+    class_dim = 102
+    dshape = [3, 224, 224] if args.order == 'NCHW' else [224, 224, 3]
+    input = fluid.layers.data(name='data', shape=dshape, dtype='float32')
+    label = fluid.layers.data(name='label', shape=[1], dtype='int64')
+    predict = model(input, class_dim)
+    cost = fluid.layers.cross_entropy(input=predict, label=label)
+    avg_cost = fluid.layers.mean(x=cost)
+    optimizer = fluid.optimizer.Momentum(learning_rate=0.01, momentum=0.9)
+    opts = optimizer.minimize(avg_cost)
+    accuracy = fluid.evaluator.Accuracy(input=predict, label=label)
+
+    train_reader = paddle.batch(
+        paddle.reader.shuffle(
+            paddle.dataset.flowers.train(), buf_size=5120),
+        batch_size=args.batch_size)
+
+    if args.device == 'CPU':
+        place = fluid.CPUPlace()
+    else:
+        place = fluid.GPUPlace(0)
+    exe = fluid.Executor(place)
+    exe.run(fluid.default_startup_program())
+
+    iter = 0
+    for pass_id in range(args.pass_num):
+        accuracy.reset(exe)
+        if iter == args.iterations:
+            break
+        for data in train_reader():
+            if iter == args.iterations:
+                break
+            image = np.array(map(lambda x: x[0].reshape(dshape), data)).astype(
+                'float32')
+            label = np.array(map(lambda x: x[1], data)).astype('int64')
+            label = label.reshape([-1, 1])
+            loss, acc = exe.run(fluid.default_main_program(),
+                                feed={'data': image,
+                                      'label': label},
+                                fetch_list=[avg_cost] + accuracy.metrics)
+            pass_acc = accuracy.eval(exe)
+            print("Iter: %d, loss: %s, acc: %s, pass_acc: %s" %
+                  (iter, str(loss), str(acc), str(pass_acc)))
+            iter += 1
+
+    duration = time.time() - start_time
+    examples_per_sec = args.iterations * args.batch_size / duration
+    sec_per_batch = duration / args.batch_size
+
+    print('\nTotal examples: %d, total time: %.5f' %
+          (args.iterations * args.batch_size, duration))
+    print('%.5f examples/sec, %.5f sec/batch \n' %
+          (examples_per_sec, sec_per_batch))
+
+    if args.use_cprof:
```

Comment:
Maybe it will be better also write stats to file.

Reply:
If need, we can add it later.